faker/providers/misc/__init__.py

```diff
@@ -27,6 +27,15 @@ def null_boolean(cls):
         }[random.randint(-1, 1)]
 
     @classmethod
+    def binary(cls, length=(1 * 1024 * 1024)):
+        """ Returns random binary blob.
+
+        Default blob size is 1 Mb.
+        """
+        return bytes(bytearray([
+            random.randrange(0, 256) for _ in range(length)]))
```

Comment:
As you wish. Did the change in commit https://github.com/kdeldycke/faker/commit/7fcec3c812d71722d8501822681f27be6c4422f4